src/home/getters/get-github-user.ts

```diff
@@ -30,13 +28,22 @@ async function getNewSessionToken(): Promise<string | null> {
   const params = new URLSearchParams(hash.substr(1)); // remove the '#' and parse
   const providerToken = params.get("provider_token");
   if (!providerToken) {
-    return null;
+    const error = params.get("error_description");
+    // supabase auth provider has failed for some reason
+    console.error(`GitHub login provider: ${error}`);
   }
-  return providerToken;
+  return providerToken || null;
 }
 
-async function getNewGitHubUser(providerToken: string): Promise<GitHubUser> {
+async function getNewGitHubUser(providerToken: string | null): Promise<GitHubUser | null> {
   const octokit = new Octokit({ auth: providerToken });
-  const response = (await octokit.request("GET /user")) as GitHubUserResponse;
-  return response.data;
+  try {
+    const response = (await octokit.request("GET /user")) as GitHubUserResponse;
+    return response.data;
+  } catch (error) {
+    if (error instanceof RequestError && error.status === 403) {
+      await handleRateLimit(providerToken ? octokit : undefined, error);
```

Comment:
```suggestion
      await handleRateLimit(providerToken ?? octokit, error);
```
Is this more concise?

Reply:
Although if there is a `providerToken` and we get an error it'll pass `providerToken` as an arg instead of `octokit` which would mean rebuilding `octokit` within `handleRateLimit()` while both invocations of `handleRateLimit()` already have `octokit` built, one is guaranteed to have a provider token and this one in question is not.

If we make this more concise we litter the rate limit function

I guess `if(typeof octokit == "string")` would work instead of `if(octokit){}else{}` but typing octokit as string seemed daft when it first popped into my head 😂